scripts/release_scripts/deploy.py

```diff
@@ -464,6 +463,34 @@ def check_release_doc():
     common.ask_user_to_confirm(message)
 
 
+def update_configs_in_deploy_data(
+        app_name, release_dir_path, deploy_data_path, personal_access_token):
+    """Updates feconf and constants file in deploy data to match the config
+    files in release scripts.
+
+    Args:
+        app_name: str. The name of the app to deploy.
+        release_dir_path: str. Path of directory where all files are copied
+            for release.
+        deploy_data_path: str. Path for deploy data directory.
+        personal_access_token: str. The personal access token for the
+            GitHub id of user.
+    """
+    if app_name == APP_NAME_OPPIASERVER:
+        update_configs.update_feconf_for_main_server(
+            release_dir_path, personal_access_token)
+
+        release_feconf_path = os.path.join(release_dir_path, common.FECONF_PATH)
```

Comment:
Probably this should be part of update_feconf_for_main_server()?

Reply:
Updated the code with a prompt, this function is not required now.

Reply:
Shouldn't this part still go in update_feconf.main()? It should be run whenever the boolean is equal to True, right?

Reply:
The functions called within this are directly called from main whenever the boolean is true, so this extra function isn't required.

Reply:
I don't understand -- I'm talking about lines 483 to 490. They shouldn't be here, they should be in update_configs, right?

Reply:
The lines are in if block when app_name is oppiaserver, so the prompt will be true and the check will be done after updating the config. The task of the update configs script is to update all the configs and the verification is done before deploying in the deploy script.

```
if app_name == APP_NAME_OPPIASERVER:
        update_configs.main(
            release_dir_path, deploy_data_path, personal_access_token, True)

        release_feconf_path = os.path.join(release_dir_path, common.FECONF_PATH)
        feconf_contents = python_utils.open_file(
            release_feconf_path, 'r').read()
        if ('MAILGUN_API_KEY' not in feconf_contents or
                'MAILGUN_API_KEY = None' in feconf_contents):
            raise Exception(
                'The mailgun API key must be added '
                'before deployment.')
    else: ...
```

Reply:
Updated as per offline discussion.